codyze-core/src/main/kotlin/de/fraunhofer/aisec/codyze/dsl/QueryTreeExtensions.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2025, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+@file:Suppress("CONTEXT_RECEIVERS_DEPRECATED")
+
+package de.fraunhofer.aisec.codyze.dsl
+
+import de.fraunhofer.aisec.cpg.query.*
+
+/** Performs a logical and (&&) operation between the values and creates and returns [Decision]s. */
+infix fun QueryTree<Boolean>.and(other: Decision): Decision {
```

Comment:
To be more complete and consistent, we may also want to make available to following combinations:
* `Boolean` x `Boolean` -> `Decision`
* `QueryTree<Boolean>` x `Boolean` -> `Decision`
* `Boolean` x `Decision` -> `Decision`
* `QueryTreey<Boolean>` x `Boolean` -> `Decision`
* `Boolean` x `QueryTree<Boolean>` -> `Decision`

or is this logic directly provided by the different `requirement` implementations in the DSL?

Reply:
> To be more complete and consistent, we may also want to make available to following combinations:
> 
> * `Boolean` x `Boolean` -> `Decision`
> * `QueryTree<Boolean>` x `Boolean` -> `Decision`
> * `Boolean` x `Decision` -> `Decision`
> * `QueryTreey<Boolean>` x `Boolean` -> `Decision`
> * `Boolean` x `QueryTree<Boolean>` -> `Decision`
> 
> or is this logic directly provided by the different `requirement` implementations in the DSL?

I hope so ^^ not sure ^^